### PR TITLE
Week 2 - Section 3 - Dictionaries

### PR DESCRIPTION
**Pull request by ksauhen**

### CODE REVIEW

File: weeks/week02/section03.md [hunk 1]
```diff
@@ -1,33 +1,222 @@
...
+```python
+results = {n: n ** 2 for n in range(10)}
+```
 
 ### Hands-On Task:
+1. [TI] How can you merge two dictionaries in Python?
+2. [TI] Is it possible to use a tuple as a key in the dictionary?
+3. Write a function that takes a dictionary and returns a list of all the keys.
+4. Created a nested dictionary.
+5. Create a shallow copy of a nested dictionary. Modify the original dictionary. How has the shallow copy changed?
```
Comment (SSN-SPB) [inline weeks/week02/section03.md:220]: It is not relevant to Dictionary but if we mention deep and shallow copy - imho we need to mention it in material (or even more - provide example of it)

**Review by SSN-SPB — APPROVED** [review]

File: weeks/week02/section03.md (hunk 1 above)
Comment (ksauhen) [reply weeks/week02/section03.md]: @SSN-SPB I am not sure if we need to copy the material from the section 1. I have added a ref note to make it clear. Is it ok?

File: weeks/week02/section03.md [hunk 2]
```diff
@@ -1,33 +1,222 @@
 # Section 3: Dictionaries
 
-Dictionaries are mutable mappings of key-value pairs, perfect for storing related data.
+A dictionary is a mutable mapping object of key-value pair items, perfect for storing related data and efficient data 
+retrieval. A dictionary can grow and shrink. It may be nested and contain other collection data types and other 
+dictionaries. Unlike sequences, which are indexed by a range of numbers, dictionaries are indexed by keys.
+
+An object of any immutable type can be used as a dictionary key. Duplicate key are not allowed. By contrast,
```
[the remark this answers is not in the document]
Comment (SSN-SPB) [reply weeks/week02/section03.md:7]: @ksauhen - yes, It sounds good to me.

File: weeks/week02/section03.md [hunk 3]
```diff
@@ -1,33 +1,222 @@
...
 2. [Python Dictionaries - Real Python](https://realpython.com/python-dicts/)
+3. [Mapping Types - dict - Python Documentation](https://docs.python.org/3/library/stdtypes.html#mapping-types-dict)
 
 ---
 
+The main operations on a dictionary are storing a value with some key and extracting the value given the key. 
 
-### 3.1 Working with Dictionaries
-
-Dictionaries allow you to store data values in key-value pairs. This structure enables efficient data retrieval.
```
[the remark this answers is not in the document]
Comment (SSN-SPB) [reply weeks/week02/section03.md:16]: @ksauhen Got it thanks.

File: weeks/week02/section03.md (hunk 1 above)
Comment (SSN-SPB) [reply weeks/week02/section03.md]: Agreed. week02/weeks/week02/section01.md It has this information.

File: weeks/week02/section03.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (ksauhen) [reply weeks/week02/section03.md:7]: @SSN-SPB As discussed, we will keep the current wording

File: weeks/week02/section03.md [hunk 4]
```diff
@@ -1,33 +1,222 @@
...
+
+In an example below we will generate a dictionary containing key n mapped to value n ** 2:
+
+### Example:
+
+```python
+results = {n: n ** 2 for n in range(10)}
+```
 
 ### Hands-On Task:
```
[the remark this answers is not in the document]
Comment (ksauhen) [reply weeks/week02/section03.md]: This was done in prevoius commit 
image

**Review by ericrommel — APPROVED** [review]
LGTM

**Pull request merged**